### DISCUSSION TITLE
Missmatch Authentication

### DISCUSSION THREAD

**Discussion by septijo** [dimaskiddo/go-whatsapp-multidevice-rest#46] [Q&A] [answered]
If we use wrong number on Login, first it will be Ok .. but seconds later it will be "cut" automatically and "Logging out WhatsApp Client Due to "Missmatch" Authentication" appear.

( already search the "Missmatch" word in source code, with not luck.. maybe the typo comes from WhatsApp ?? hahaha )

How to detect it ? please add one more API end point, such as /checkLoggedin

I understand that the "/login" end point can not wait for correct respond, because it return QR Code and must wait for the scan process.

thanks a lot Mr. Dimas

**Comment by dimaskiddo** [comment, marked as answer]
Sorry @septijo for my late response and sorry to revert your question back to make everyone also can understand whats's exactly happened

So i add some feature that is related to security issue when restoring whatsapp multidevice and multi-session when the server is restarted. If you put wrong username as JIDs then when it's restored it will be changed to the correct JIDs. So to prevent that i put a self cron system to check every registered JIDs with its authentication identifier mapping every minute, when it's not in the same data then it will force it to logout, so when the server is restarted and restoring back the session it will mapped correctly with the identifier mapping

I hope my answer can help 😄 🥂
